server/tests/e2e/federated.spec.ts

```diff
@@ -163,10 +163,153 @@ describe("end-to-end federated", () => {
     const [m1, m2] = await Promise.all([lusCovidUser(), lusCovidUser()]);
     assert.isTrue(m1.equals(m2));
   });
-
-  it("trains wikitext", async function () {
+  
+  it("two wikitext reach consensus", async function () {
     this.timeout("3m");
+    
+    const [m1, m2] = await Promise.all([wikitextUser(), wikitextUser()]);
+    assert.isTrue(m1.equals(m2))
+  });
+
+  it("clients emit expected statuses", async function () {
+    this.timeout(15_000);
+    const lusCovidTask = defaultTasks.lusCovid.getTask();
+    lusCovidTask.trainingInformation.epochs = 8;
+    lusCovidTask.trainingInformation.roundDuration = 2;
+    lusCovidTask.trainingInformation.minNbOfParticipants = 2;
+
+    const [positive, negative] = [
+      (
+        await loadImagesInDir(path.join(DATASET_DIR, "lus_covid", "COVID+"))
+      ).zip(Repeat("COVID-Positive")),
+      (
+        await loadImagesInDir(path.join(DATASET_DIR, "lus_covid", "COVID-"))
+      ).zip(Repeat("COVID-Negative")),
+    ];
+    const dataset = positive.chain(negative);
 
-    await wikitextUser();
+    /**
+     * When disco.trainByRound is called for the first time, the client connects to the server
+     * which returns the latest model, current round and nb of participants.
+     * Then at each round the event cycle is:
+     * a) onRoundBeingCommunication which updates the status to TRAINING
+     * b) local training (the status remains TRAINING)
+     * c) onRoundEndCommunication which sends the local update and 
+     * receives the global weights while emitting the status UPDATE
+     * 
+     * Given this, it is important to note that calling disco.trainByRound().next()
+     * for the first time will perform a) and then b) where it stops and yields the round logs.
+     * Thus, c) isn't done and the model aggregation by the server is not performed during this first call to next().
+     * 
+     * Calling next() again will then do c), and back to a) and b).
+     * 
+     * In this test the timeline is:
+     * - User 1 joins the task by themselves
+     * - User 2 joins
+     * - User 1 leaves
+     * - User 3 joins
+     * - User 2 & 3 leave
+     */
+    const TRAINING: RoundStatus = "Training the model on the data you connected"
+    const WAITING: RoundStatus = "Waiting for more participants"
+    const UPDATING: RoundStatus = "Updating the model with other participants' models"
+    const statusUpdateTime = 500
+
+    // Create User 1
+    const discoUser1 = new Disco(lusCovidTask, url, { scheme: "federated" });
+    let statusUser1: RoundStatus | undefined;
+    discoUser1.on("status", status => {
+      statusUser1 = status
+      console.log("user 1 status", status)
+    })
+    const generatorUser1 = discoUser1.trainByRound(["image", dataset])
+    
+    // Have User 1 join the task and train locally for one round
+    const logUser1Round1 = await generatorUser1.next()
+    if (logUser1Round1.done) throw Error("User 1 finished training at their 1st round")
```

Comment:
makes for nicer error message

```suggestion
    expect(logUser1Round1.done).to.be.false
```